pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -309,12 +310,34 @@ func (service *SecurityPolicyService) buildExpressionsMatchExpression(matchExpre
 	return err
 }
 
-func (service *SecurityPolicyService) buildPolicyGroupID(obj *v1alpha1.SecurityPolicy) string {
+// build appliedTo group ID for both policy and rule levels.
+func (service *SecurityPolicyService) buildAppliedGroupID(obj *v1alpha1.SecurityPolicy, ruleIdx int) string {
+	if ruleIdx != -1 {
+		return fmt.Sprintf("sp_%s_%d_scope", obj.UID, ruleIdx)
+	}
+
 	return fmt.Sprintf("sp_%s_scope", obj.UID)
 }
 
-func (service *SecurityPolicyService) buildPolicyGroupPath(obj *v1alpha1.SecurityPolicy) string {
-	return fmt.Sprintf("/infra/domains/%s/groups/%s", getDomain(service), service.buildPolicyGroupID(obj))
+// build appliedTo group path for both policy and rule levels.
+func (service *SecurityPolicyService) buildAppliedGroupPath(obj *v1alpha1.SecurityPolicy, groupID string) string {
```

Comment:
How do you think about unifying the styles for these 3 functions `buildAppliedGroupID`, `buildAppliedGroupName` and `buildAppliedGroupPath`? i.e.
```
func (service *SecurityPolicyService) buildAppliedGroupId(obj *v1alpha1.SecurityPolicy, ruleIdx int) string {
    ...
}
...
func (service *SecurityPolicyService) buildAppliedGroupName(obj *v1alpha1.SecurityPolicy, ruleIdx int) string {
    ...
}
...
func (service *SecurityPolicyService) buildAppliedGroupPath(obj *v1alpha1.SecurityPolicy, ruleIdx int) string {
    ...
}
```
And the similar question for `buildRulePeerGroupXXX`

Reply:
Thanks Qian. That's a good idea.  I refactored the code to unify the buildAppliedGroupID, buildAppliedGroupName and buildAppliedGroupPath as you suggested.